aptos-move/framework/supra-framework/sources/poel/poel.move

```diff
@@ -0,0 +1,308 @@
+///
+/// This is the Proof of Efficiency Liquidity module used to delagate $supra to staking pools for users
+/// who hold iEth by making deposits of ETH to the intralayer vault. It handles the creation of the iETH and distribution to the
+/// corresponding addresses on supra
+///
+/// Flow:
+/// 1. Users submit WETH to the Intralayer Vault on the Ethereum blockchain.
+/// 2. The Supra cross-chain communication protocol transfers the deposit information to the PoEL (Proof of Efficient Liquidity) contract on the Supra chain.
+/// 3. This process creates a new token, iETH (InterLayer ETH), which is distributed to the user's address on Supra. 
+/// 4. In the background, utilizing the PoEL protocol, assets deposited into the Intralayer Vault serve as collateral to 
+/// borrow $Supra and stake it in the Supra PoS system. The PoEL contract periodically lends $Supra tokens to users who have submitted WETH to the Intralayer 
+/// Vault and evenly distributes these tokens among various delegation pools participating in the system. 
+/// In this setup, the PoEL contract itself acts as the delegator of the $Supra tokens in the delegation pool on behalf of the users.
+/// Note:
+/// The total lended amount through the PoEL contract depends on the Oracle price submitted to the PoEL contract.
+/// If the price of the asset goes up compared with $Supra the system lends more delegating more assets 
+/// to the delegation pools and correspondingly if the price drops the system lends less, withdrawing some assets
+/// from the delegation pools. Therefore, the PoEL contract acting as the delegation agent for the users ensures the
+/// total delegated amount from the PoEL contract is equable with the total nominal value submitted to the IntraLayer vaults.
+/// When a user wants to retrieve the original ETH that was used as collateral, they can burn the iETH, which then enables
+/// them to receive the original ETH on the Ethereum blockchain.
+
+module supra_framework::poel {
+    use supra_framework::table;
+    use supra_framework::smart_table::SmartTable;
+    //Errors:
+
+    /// thrown when the allowed signer is not the caller (owner)
+    const ERR_NOT_OWNER: u64 = 0;
+
+    /// thrown when the wrong asset weight gets calculated
+    const ERR_WRONG_WEIGHT: u64 = 1;
+
+    /// thrown when the length of the supplied collaterixation weight argument vector does not match table entries
+    const ERR_WRONG_CWV_LENGTH: u64 = 2;
+
+    /// thrown when the length of the desirability score argument vector does not match table entries
+    const ERR_WRONG_DESIRABILITY_SCORE_LEN: u64 = 3;
+
+    /// thrown when the asset id supiled is not present in the TotalLiquidityTable
+    const ERR_ASSET_NOT_PRESENT: u64 = 4;
+
+    /// thrown If coefficient_X < coefficient_b
+    const ERR_COEFA_LT_COEFB: u64 = 5;
+ 
+    // Structs:
```

Comment:
DO NOT use `0` for error code. Also the convention is just to use `E` instead of `ERR` as prefix for error code.

Reply:
Okay. Will update that.